plugins/modules/syslog_server.py

```diff
@@ -0,0 +1,320 @@
+    new_entry = api_val
+    if param_val and (param_val != default and param_val != api_val):
+        new_entry = param_val
+    return new_entry
+
+
+def build_update_payload(
+    module: AnsibleModule, rest_client: RestClient, syslog_server: SyslogServer
+):
+    payload = {
+        "host": syslog_server.host,
+        "port": syslog_server.port,
+        "protocol": syslog_server.protocol,
+    }
+    if module.params["host_new"] and syslog_server.host != module.params["host_new"]:
+        payload["host"] = module.params["host_new"]
+    if (
+        module.params["port"] != DEFAULT_PORT
+        and module.params["port"] != syslog_server.port
+    ):
+        payload["port"] = module.params["port"]
+    if module.params["protocol"]:
+        protocol = get_protocol(module.params["protocol"])
+        if protocol != UDP and protocol != syslog_server.protocol:
+            payload["protocol"] = protocol
+    return payload
+
+
+def update_syslog_server(module: AnsibleModule, rest_client: RestClient):
+    # Get record of old syslogServer by host
+    old_syserver = SyslogServer.get_by_host(
+        host=module.params["host"], rest_client=rest_client
+    )
+
+    before = old_syserver.to_ansible()
+    payload = build_update_payload(module, rest_client, old_syserver)
+
+    module.warn("payload: " + str(payload))
+
+    # Return if there are no changes
+    if payload == {
+        "host": before.get("host"),
+        "port": before.get("port"),
+        "protocol": before.get("protocol"),
+    }:
+        return False, before, dict(before=before, after=before)
+
+    # Otherwise, update with new email address
+    old_syserver.update(
+        rest_client=rest_client,
+        payload=payload,
+        check_mode=module.check_mode,
+    )
+
+    new_syserver = SyslogServer.get_by_host(
+        host=payload["host"], rest_client=rest_client
+    )
+    after = new_syserver.to_ansible()
+
+    return (
+        after != before,
+        after,
+        dict(before=before, after=after),
+    )  # changed, records, diff
+
+
+def delete_syslog_server(module: AnsibleModule, rest_client: RestClient):
+    delete_syserver = SyslogServer.get_by_host(
+        host=module.params["host"], rest_client=rest_client
+    )
+
+    if not delete_syserver:
+        return False, {}, dict(before={}, after={})
+
+    before = delete_syserver.to_ansible()
+    delete_syserver.delete(rest_client, module.check_mode)
+
+    return (
+        True,
+        {},
+        dict(before=before, after={}),
+    )  # changed, records, diff
+
+
+def run(module: AnsibleModule, rest_client: RestClient):
+    syslog_server = SyslogServer.get_by_host(
+        host=module.params["host"], rest_client=rest_client
+    )
+    state = module.params["state"]
+    if state == "present":
+        if syslog_server:
+            return update_syslog_server(module, rest_client)
+        return create_syslog_server(module, rest_client)
```

Comment:
Please pass the `syslog_server` to both `update_syslog_server` and `create_syslog_server`. They both call `SyslogServer.get_by_host` as the very first thing.

Looking a this a second time. What happens when we try to rename `host` to `host_new`? What reports integration test - does task 'name: Update Syslog server - HOST ONLY' pass?

Reply:
Yes, task 'name: Update Syslog server - HOST ONLY' passes, the idempotence of this task doesn't...

Reply:
I think I've figured out why exactly the idempotence of this task fails.
In task 'Update Syslog server - HOST ONLY', a syslog server host "0.0.0.0" is changed to "1.2.3.4" which makes server host "0.0.0.0" nonexistent. And then in the next task 'Update Syslog server - HOST ONLY - idempotence' it again tries to update syslog server host "0.0.0.0" to "1.2.3.4" but because that host ("0.0.0.0") doesn't exist anymore, it just creates one, hence why that test fails, when asserting ``result.changed == False`` and ``result.diff.before == result.diff.after``.